de.metas.business/src/main/java/org/adempiere/impexp/product/ProductPriceRequest.java

```diff
@@ -0,0 +1,53 @@
+/**
+ *
+ */
+package org.adempiere.impexp.product;
+
+import java.math.BigDecimal;
+import java.sql.Timestamp;
+
+import org.compiere.model.I_C_TaxCategory;
+import org.compiere.model.I_M_PriceList;
+
+import lombok.Builder;
+import lombok.Getter;
+import lombok.NonNull;
+
+/*
+ * #%L
+ * de.metas.business
+ * %%
+ * Copyright (C) 2018 metas GmbH
+ * %%
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation, either version 2 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program.  If not, see
+ * <http://www.gnu.org/licenses/gpl-2.0.html>.
+ * #L%
+ */
+
+/**
+ * @author metas-dev <dev@metasfresh.com>
+ *
+ */
+@Builder
+@Getter
+public class ProductPriceRequest
+{
+	private final org.compiere.model.I_M_Product product;
+	private final I_M_PriceList priceList;
+	@NonNull
+	private final BigDecimal price;
+	final Timestamp validDate;
+	@NonNull
+	final I_C_TaxCategory taxCategory;
+}
```

Comment:
@cristinamghita  @metas-ts 
* all of those shall be IDs
* `ProductPriceRequest` says nothing. There is `predicate` inside it. Maybe we shall call it `ProductPriceCreateRequest` in case we are talking about creating product prices

for future: we shall start using LocalDate instead of Timestamp. I've already started to add converters in TimeUtil.

Reply:
another thing: please decide if u writing `private final` or just `final` or just `private` but all options together in same class.
anyways, when annotated with `@Value`, lombok will make everything `private final`.

Reply:
> all of those shall be IDs

agree. imho it would be even better to have e.g. a PriceList pojo with the required data..
with that
* we even would not have to rely the "the other side" is able to load the record
* we know which columns of the pricelist are supposed to be relevant

but i'm fine with leaving that imho-improvement to the future